package/gluon-client-bridge/luasrc/lib/gluon/upgrade/320-gluon-client-bridge-wireless

```diff
@@ -42,6 +38,73 @@ util.foreach_radio(uci, function(radio, index, config)
 		ifname = suffix and 'client' .. suffix,
 		disabled = disabled or false,
 	})
+end
+
+local function configure_owe(radio, index, config, radio_name)
+	local name_client = 'client_' .. radio_name
+	local name_owe = 'owe_' .. radio_name
+	local suffix = radio_name:match('^radio(%d+)$')
+
+	local ap = config.ap
+	local owe = config.owe
+
+	local disabled = uci:get_bool('wireless', name_client, 'disabled')
+
+	uci:delete('wireless', name_owe)
+	local macaddr = util.get_wlan_mac(uci, radio, index, 3)
+
+	if not owe() or not ap() or not macaddr then
+		return
+	end
+
+	uci:section('wireless', 'wifi-iface', name_owe, {
+		device = radio_name,
+		network = 'client',
+		mode = 'ap',
+		ssid = owe.ssid(),
+		macaddr = macaddr,
+		ifname = suffix and 'owe' .. suffix,
+		disabled = disabled or false,
+		encryption = 'owe',
+		mfp = 2,
+	})
+end
+
+local function configure_owe_tm(radio, index, config, radio_name)
+	local name_client = 'client_' .. radio_name
+	local name_owe = 'owe_' .. radio_name
+
+	local ssid_client = uci:get('wireless', name_client, 'ssid')
+	local ssid_owe = uci:get('wireless', name_owe, 'ssid')
+
+	local macaddr_client = uci:get('wireless', name_client, 'macaddr')
+	local macaddr_owe = uci:get('wireless', name_owe, 'macaddr')
+
+	local owe = config.owe
+
+	if not owe() or not owe.transition_mode() or not ssid_client or not ssid_owe or not macaddr_client or not macaddr_owe then
+		return
+	end
+
+	uci:set('wireless', name_client, 'owe_transition_ssid', ssid_owe)
+	uci:set('wireless', name_client, 'owe_transition_bssid', macaddr_owe)
+	
+	uci:set('wireless', name_owe, 'owe_transition_ssid', ssid_client)
+	uci:set('wireless', name_owe, 'owe_transition_bssid', macaddr_client)
+	uci:set('wireless', name_owe, 'hidden', '1')
+end
+
+util.foreach_radio(uci, function(radio, index, config)
+	local radio_name = radio['.name']
+
+	configure_ap(radio, index, config, radio_name)
+
+	if not platform.supports_wpa3() then
```

Comment:
How is `supports_wpa3()` appropriate for feature gating OWE?

Reply:
OWe requires MFP (hardware feature) as well as hostapd-openssl (software-feature).

Reply:
Maybe make that a comment.

Reply:
I've just double-checked - supports_wpa3 returns true in case a suitable hostapd is installed.

We acutally need to assert on supports_mfp here, as this implies the hostapd availability as well as all WiFi interfaces supporting MFP. This is not the case with broken draft-n Atheros chips.